Mailbox.Tests/DataLoaderTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Mailbox.Tests
+{
+    [TestClass]
+    public class DataLoaderTests
+    {
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ConstructorPassedNull()
+        {
+            DataLoader dataLoader = new DataLoader(null);
+            dataLoader.Dispose();
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void SavePassedNull()
+        {
+            MemoryStream stream = new MemoryStream();
+            DataLoader dataLoader = new DataLoader(stream);
```

Comment:
Rather than trying to dispose this manually, it would be better to dispose it with a using statement.
```suggestion
            using DataLoader dataLoader = new DataLoader(stream);
```